contracts/modules/v2/StakingPool.sol

```diff
@@ -507,6 +507,131 @@ contract StakingPool is IStakingPool, ERC721 {
+      );
+
+      return; // Done! Skip the rest of the function.
+    }
+
+    // If the initial tranche is still active, move all the shares and pending rewards to the
+    // newly chosen tranche and its coresponding deopsit.
+
+    // First make sure tranches are up to date in terms of accumulated NXM rewards.
+    updateTranches();
+
+    Tranche memory initialTranche = tranches[initialTrancheId];
+    Tranche memory newTranche = tranches[newTrancheId];
+    Deposit memory initialDeposit = deposits[tokenId][initialTrancheId];
+    Deposit memory newDeposit = deposits[tokenId][newTrancheId];
+    uint _accNxmPerRewardsShare = accNxmPerRewardsShare;
+
+
+    // The user's shares are moved from the initial tranche to the new one.
+    initialTranche.stakeShares -= initialDeposit.stakeShares;
+    initialTranche.rewardsShares -= initialDeposit.rewardsShares;
+    newTranche.stakeShares += initialDeposit.stakeShares;
+    newTranche.rewardsShares += initialDeposit.rewardsShares;
+
+    // Store the updated tranches.
+    tranches[initialTrancheId] = initialTranche;
+    tranches[newTrancheId] = newTranche;
+
+    // Calculate the rewards that will be carried from the initial deposit to the next one.
+    uint rewardsToCarry;
+    {
+      uint newEarningsPerShare = _accNxmPerRewardsShare - initialDeposit.lastAccNxmPerRewardShare;
+      rewardsToCarry = newEarningsPerShare * initialDeposit.rewardsShares
+        + initialDeposit.pendingRewards;
+    }
+
+    // Set the pending rewards from the intial deposit to zero since they will be carried over to
+    // the new one.
+    initialDeposit.pendingRewards = 0;
+
+    // If a deposit lasting until the new tranche's end date already exists, calculate its pending
+    // rewards before carrying over the rewards from the inital deposit.
+    if (newDeposit.lastAccNxmPerRewardShare != 0) {
+      uint newEarningsPerShare = _accNxmPerRewardsShare - newDeposit.lastAccNxmPerRewardShare;
+      newDeposit.pendingRewards += newEarningsPerShare * newDeposit.rewardsShares;
+    }
+
+    // The carried rewards are added to the pending rewards of the new depostit.
+    newDeposit.pendingRewards += rewardsToCarry;
+
+    // Update the last value of accumulated NXM per share in the new deposit.
+    newDeposit.lastAccNxmPerRewardShare = _accNxmPerRewardsShare;
+
+    // Reset the last value of accumulated NXM per share in the initial deposit. In case the user
+    // decides to make another deposit until the end of initial tranche, it has to be treated as a
+    // new deposit, not an increase.
+    initialDeposit.lastAccNxmPerRewardShare = 0;
+
+    // Move the user's shares from the initial deposit to the new one.
+    initialDeposit.stakeShares = 0;
+    initialDeposit.rewardsShares = 0;
+    newDeposit.rewardsShares += initialDeposit.rewardsShares;
+    newDeposit.stakeShares += initialDeposit.stakeShares;
+
+    // Store the updated deposits.
+    deposits[tokenId][initialTrancheId] = initialDeposit;
```

Comment:
All fields inside the `initialDeposit` are being zeroed out, you can use `delete deposits[tokenId][initialTrancheId]` directly and remove the lines that update `initialDeposit`'s in-memory contents